Sources/Sentry/SentryFileManager.m

```diff
@@ -335,7 +344,9 @@ - (void)storeSession:(SentrySession *)session sessionFilePath:(NSString *)sessio
     NSData *sessionData = [SentrySerialization dataWithSession:session error:nil];
     SENTRY_LOG_DEBUG(@"Writing session: %@", sessionFilePath);
     @synchronized(self.currentSessionFilePath) {
-        [sessionData writeToFile:sessionFilePath options:NSDataWritingAtomic error:nil];
```

Comment:
This crashed in a test case because the sentry directory had been deleted. This is a good thing to guard against in production, and to log if/when it does happen.

All calls to `writeData:toPath:` call the new method `ensureSentryDirectory` to do the check and log.